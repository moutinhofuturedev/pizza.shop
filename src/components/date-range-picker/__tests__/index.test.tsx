import { fireEvent, render, screen } from '@testing-library/react'
import { DatePickerWithRange } from '../index'

describe('DatePickerWithRange', () => {
  it('renders with no date', () => {
    const onDateChange = vi.fn()
    const date = { from: undefined, to: undefined }

    render(<DatePickerWithRange date={date} onDateChange={onDateChange} />)

    const calendar = screen.getByText('Selecione uma data')

    expect(calendar).toBeInTheDocument()
  })

  it('renders with a date range selected', () => {
    const onDateChange = vi.fn()
    const date = {
      from: new Date('2024-09-20T00:00:00.000Z'),
      to: new Date('2024-09-27T00:00:00.000Z'),
    }

    const { container } = render(
      <DatePickerWithRange date={date} onDateChange={onDateChange} />
    )

    const calendar = screen.getByText('Sep 19, 2024 - Sep 26, 2024')
    fireEvent.click(calendar)

    expect(calendar).toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })
})
