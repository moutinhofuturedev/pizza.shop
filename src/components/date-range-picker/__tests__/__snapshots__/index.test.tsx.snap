// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DatePickerWithRange > renders with a date range selected 1`] = `
<div>
  <div
    class="grid gap-2"
  >
    <button
      aria-controls="radix-:r1:"
      aria-expanded="true"
      aria-haspopup="dialog"
      class="inline-flex items-center whitespace-nowrap rounded-md text-sm ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 border border-input bg-background hover:bg-accent hover:text-accent-foreground h-10 px-4 py-2 w-[300px] justify-start text-left font-normal"
      data-state="open"
      id="date"
      type="button"
    >
      <svg
        class="lucide lucide-calendar mr-2 h-4 w-4"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8 2v4"
        />
        <path
          d="M16 2v4"
        />
        <rect
          height="18"
          rx="2"
          width="18"
          x="3"
          y="4"
        />
        <path
          d="M3 10h18"
        />
      </svg>
      Dec 31, 2021
       -
       
      Jan 30, 2022
    </button>
  </div>
</div>
`;
